### PR TITLE
feature: Change Join and Exit providers to use provide/inject pattern

### PR DESCRIPTION
**Pull request by simeonkerkola**
# Description

Provide Join and Exit pool logic without wrapper components

## Type of change

- [x] Code refactor / cleanup

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- Test Boosted pool joins and exits still work
  - Join: http://localhost:8080/#/polygon/pool/0x48e6b98ef6329f8f0a30ebb8c7c960330d64808500000000000000000000075b/invest
  - Exit: http://localhost:8080/#/polygon/pool/0x48e6b98ef6329f8f0a30ebb8c7c960330d64808500000000000000000000075b/withdraw
- Test other pool types joins and exits still work
  - Join: http://localhost:8080/#/polygon/pool/0x3d468ab2329f296e1b9d8476bb54dd77d8c2320f000200000000000000000426/invest
  - Exit: http://localhost:8080/#/polygon/pool/0x3d468ab2329f296e1b9d8476bb54dd77d8c2320f000200000000000000000426/withdraw

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/components/contextual/pages/pool/invest/InvestPageInvestBlock.vue [hunk 1]
```diff
@@ -19,6 +23,10 @@ const { pool, loadingPool, transfersAllowed } = usePoolTransfers();
 const { activeTab, resetTabs } = useInvestPageTabs();
 const { isDeepPool, isPreMintedBptPool } = usePool(pool);
 
+/**
+ * PROVIDERS
+ */
```
Comment (garethfuller) [inline src/components/contextual/pages/pool/invest/InvestPageInvestBlock.vue:28]: Remove?

File: src/pages/pool/invest.vue [hunk 2]
```diff
@@ -37,13 +32,7 @@ useIntervalFn(poolQuery.refetch, oneMinInMs);
...
 <template>
-  <JoinPoolProvider
-    v-if="pool && isDeepPool"
-    :pool="pool"
-    :isSingleAssetJoin="activeTab === Tab.SingleToken"
-  >
-    <InvestPage></InvestPage>
-  </JoinPoolProvider>
-  <InvestPage v-else></InvestPage>
+  <!-- TODO: handle loading state -->
```
Comment (garethfuller) [inline src/pages/pool/invest.vue:35]: What exactly needs to be handled?

**Review by garethfuller — APPROVED** [review]
Tested multiple joins/exits and all seem to be working.

File: src/pages/pool/invest.vue (hunk 2 above)
Comment (simeonkerkola) [reply src/pages/pool/invest.vue]: Had to move the loading state here, so `provideJoinPool` can be called in the child: `InvestPage.vue`

**Pull request merged**